api/v1beta1/imagerepository_types.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// / ImageRepositorySpec defines the desired state of ImageRepository
+type ImageRepositorySpec struct {
+	// Requested image repository configuration.
+	// +optional
+	Image ImageParameters `json:"image,omitempty"`
+
+	// Credentials management.
+	// +optional
+	Credentials ImageCredentials `json:"credentials,omitempty"`
```

Comment:
nit: `oc get -o yaml imagerepository.appstudio.redhat.com/imagerepository-sample`:
```
...
spec:
  credentials: {}
...
```

Reply:
That's done to make it easier to request credentials recreation. We can omit empty field. WDYT?